tools/Utils.psm1

```diff
@@ -0,0 +1,181 @@
+
+            if (loadFrom == null)
+            {
+                loadFrom = Assembly.ReflectionOnlyLoadFrom(fullAssemblyPath);
+            }
+
+            return loadFrom.GetExportedTypes().ToList();
+        }
+
+        public static List<Type> GetActivatableTypesFromFile(string assemblyPath, string referenceWinmds)
+        {
+            var exportedTypes = FromFile(assemblyPath, referenceWinmds);
+            List<Type> activatableTypes = new List<Type>();
+            foreach(var type in exportedTypes)
+            {
+                var attributes = CustomAttributeData.GetCustomAttributes(type);
+                foreach(var attrib in attributes)
+                {
+                    var attributeString = attrib.ToString();
+                    if(attributeString.Contains("Windows.Foundation.Metadata.ComposableAttribute") ||
+                       attributeString.Contains("Windows.Foundation.Metadata.ActivatableAttribute") ||
+                       attributeString.Contains("Windows.Foundation.Metadata.StaticAttribute"))
+                    {
+                        activatableTypes.Add(type);
+                        break;
+                    }   
+                }
+            }
+
+            return activatableTypes;
+        }
+    }
+}
+"@
+
+function Get-ActivatableTypes
+{
+    param(
+        [Parameter(Mandatory=$true, HelpMessage="Paths to the WinMD files whose types need to be added to the manifest.")]
+        [string]$WinmdPaths,
+
+        [Parameter(Mandatory=$true, HelpMessage="Semicolon-delimited list of reference WinMD files.")]
+        [string]$ReferenceWinmds
+    )
+
+    $winmdPathList = $WinmdPaths.Split(";", [System.StringSplitOptions]::RemoveEmptyEntries)
+    [System.Collections.Generic.List[System.Type]]$typeList = @()
+
+    foreach ($winmdPath in $winmdPathList)
+    {
+        $typeList.AddRange([WinmdHelper.LoadWinmdTypes]::GetActivatableTypesFromFile($winmdPath, $ReferenceWinmds))
+    }
+
+    $typeList
+}
+
+function Get-WinmdTypes
+{
+    param(
+        [Parameter(Mandatory=$true, HelpMessage="Paths to the WinMD files whose types need to be added to the manifest.")]
+        [string]$WinmdPaths,
+
+        [Parameter(Mandatory=$true, HelpMessage="Semicolon-delimited list of reference WinMD files.")]
+        [string]$ReferenceWinmds
+    )
+
+    $winmdPathList = $WinmdPaths.Split(";", [System.StringSplitOptions]::RemoveEmptyEntries)
+    [System.Collections.Generic.List[System.Type]]$typeList = @()
+
+    foreach ($winmdPath in $winmdPathList)
+    {
+        $typeList.AddRange([WinmdHelper.LoadWinmdTypes]::FromFile($winmdPath, $ReferenceWinmds))
+    }
+
+    $typeList
+}
+
+function Get-WindowsNamespace
+{
+    param(
+        [Parameter(Mandatory=$true)]
+        [string]$Namespace
+    )
+#UNDONE - revisit what this does and rename to Microsoft.ProjectReunion, etc. 
+    $Namespace.Replace("Microsoft.UI.Xaml", "Windows.UI.Xaml").Replace("Microsoft.UI.Private", "MUXControls")
```

Comment:
looks unused even in XAML. Remove it?

Reply:
coordinate removal with xaml? (I sent mail about this).

Reply:
I changed the interior of this method into a throw statement, we can remove later.